### PR TITLE
Upgrade efivar to version 39

### PR DESCRIPTION
**Pull request by fintelia**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Upgrade efivar to version 39

###### Change Log  
- Spec upgrade and add two new build dependencies
 
###### Does this affect the toolchain?  

**NO**

###### Test Methodology
- Pipeline build id: 509801, 510471

### CODE REVIEW

File: SPECS/efivar/efivar.spec [hunk 1]
```diff
@@ -52,6 +54,9 @@ It contains the libraries and header files to create applications
 %{_mandir}/man3/*
 
 %changelog
+* Tue Feb 06 2024 Cameron Baird <cameronbaird@microsoft.com> - 39-1
```
Comment (ndubchak) [inline SPECS/efivar/efivar.spec:57]: Update the changelog?

**Review by ndubchak — APPROVED** [review]

File: SPECS/efivar/efivar.spec [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-Release:        6%{?dist}
+Version:        39
+Release:        1%{?dist}
 License:        LGPL-2.1
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          System Environment/System Utilities
 URL:            https://github.com/rhboot/efivar
-Source0:        https://github.com/rhboot/efivar/releases/download/%{version}/%{name}-%{version}.tar.bz2
+Source0:        https://github.com/rhboot/efivar/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
```
Comment (ndubchak) [inline SPECS/efivar/efivar.spec:10]: I noticed that the updated Source0 is different from the new downloadUrl; is that correct?

Comment (fintelia) [reply SPECS/efivar/efivar.spec]: No. I think the `#` was meant to start a comment, but it actually ended up making an irrelevant change to the URL...

Comment (fintelia) [reply SPECS/efivar/efivar.spec:10]: Actually, I think it was right. The `#` caused the downloaded file to have the name `efivar-39.tar.gz` which is needed for the blob storage

**Pull request merged**